package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -185,7 +185,7 @@ end
 -- 6: owe1
 -- 7: wan_radio1 (private WLAN); mesh VPN
 function M.generate_mac(i)
-	if i > 7 or i < 0 then return nil end -- max allowed id (0b111)
```

Comment:
this is a hack because v4 mesh requires IPs for each interface. we'd either need something proper or just live with it.

Reply:
We tried to rework the MAC address generation logic in https://github.com/freifunk-gluon/gluon/pull/1983 to provide more addresses,

Reply:
In the babel implementation we just assume that we are using ipv6-only. This *could* be a valid assumption for olsr as well.
This would mean that the ipv4-only bits of the internet can be reached by implementing xlat.
In our babel net we use jool and on the nodes the xlat helpers. 
It would save a royal headache that would arise from handling ipv4 addresses in the mesh.
What do you think?